src/core/js/app.js

```diff
@@ -1,225 +1,257 @@
 
-            mapAdaptIdsToObjects();
+      mapAdaptIdsToObjects();
 
-            Adapt.log.debug('Firing app:dataLoaded');
+      Adapt.log.debug('Firing app:dataLoaded');
 
-            try {
-                Adapt.trigger('app:dataLoaded');// Triggered to setup model connections in AdaptModel.js
-            } catch(e) {
-                Adapt.log.error('Error during app:dataLoading trigger', e);
-            }
+      try {
+        Adapt.trigger('app:dataLoaded');// Triggered to setup model connections in AdaptModel.js
+      } catch(e) {
+        Adapt.log.error('Error during app:dataLoading trigger', e);
+      }
 
-            Adapt.setupMapping();
+      Adapt.setupMapping();
 
-            Adapt.wait.queue(function() {
-                triggerDataReady(newLanguage);
-            });
+      Adapt.wait.queue(function() {
+        triggerDataReady(newLanguage);
+      });
 
+    }
+  };
+
+  function triggerDataReady(newLanguage) {
+    if (newLanguage) {
+
+      Adapt.trigger('app:languageChanged', newLanguage);
+
+      _.defer(function() {
+        var startController = new StartController();
+        var hash = '#/';
+
+        if (startController.isEnabled()) {
+          hash = startController.getStartHash(true);
         }
-    };
 
-    function triggerDataReady(newLanguage) {
-        if (newLanguage) {
+        Backbone.history.navigate(hash, { trigger: true, replace: true });
+      });
+    }
+
+    Adapt.log.debug('Firing app:dataReady');
 
-            Adapt.trigger('app:languageChanged', newLanguage);
+    try {
+      Adapt.trigger('app:dataReady');
+    } catch(e) {
+      Adapt.log.error('Error during app:dataReady trigger', e);
+    }
 
-            _.defer(function() {
-                var startController = new StartController();
-                var hash = '#/';
+    Adapt.wait.queue(triggerInitialize);
 
-                if (startController.isEnabled()) {
-                    hash = startController.getStartHash(true);
-                }
+  }
 
-                Backbone.history.navigate(hash, { trigger: true, replace: true });
-            });
-        }
+  function triggerInitialize() {
+    Adapt.log.debug('Calling Adapt.initialize');
+
+    addDirection();
+    disableAnimation();
+    addNavigationBar();
+
+    Adapt.initialize();
 
-        Adapt.log.debug('Firing app:dataReady');
+    Adapt.off('adaptCollection:dataLoaded courseModel:dataLoaded');
+  }
 
-        try {
-            Adapt.trigger('app:dataReady');
-        } catch(e) {
-            Adapt.log.error('Error during app:dataReady trigger', e);
+  function addDirection() {
+    var defaultDirection = Adapt.config.get('_defaultDirection');
+
+    if (defaultDirection === 'rtl') {
+      $('html').addClass('dir-rtl').attr('dir', 'rtl');
+    } else {
+      $('html').addClass('dir-ltr').attr('dir', 'ltr');
+    }
+  }
+
+  function disableAnimation() {
+    var disableAnimationArray = Adapt.config.get('_disableAnimationFor');
```

Comment:
this is an excellent suggestion but I feel that an improvement like this might be better suited to https://github.com/adaptlearning/adapt_framework/pull/2514 (as improving that 'entry point' code is what that PR is all about). It's going to need a fair bit of re-work once all this is merged anyway so wouldn't be too much trouble to add something like this into the pile!

Reply:
Looks like #2514 has been and gone...@guywillis, can this still be actioned?

Reply:
@tomgreenfield It doesn't look like the suggest code amend above supports the `_disableAnimation` config that can be used to turn off animation globally. Could you amend your code suggestion to support it please?